scripts/github-comment/Sources/github-comment/DiffParser.swift

```diff
@@ -0,0 +1,113 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+
+struct Hunk {
+  let beforeRange: Range<Int>
+  let afterRange: Range<Int>
+  let contents: [String]
+}
+
+/**
+ Returns a pair of dictionaries of filenames to lists of hunks from a given diff format.
+
+ First dictionary are the "before" files.
+ Second dictionary are the "after" files.
+
+ // Before and after may match if no files were moved added or deleted.
+ */
+func hunksFromDiff(_ diffString: String) -> (before: [String: [Hunk]], after: [String: [Hunk]]) {
+  var beforeHunks: [String: [Hunk]] = [:]
+  var afterHunks: [String: [Hunk]] = [:]
+
+  var currentBeforeFile: String? = nil
+  var currentAfterFile: String? = nil
+  var currentBeforeRange: Range<Int>? = nil
+  var currentAfterRange: Range<Int>? = nil
+  var contents: [String] = []
+  let commitHunkIfPossible = {
+    guard let beforeFile = currentBeforeFile,
+        let afterFile = currentAfterFile,
+        let beforeRange = currentBeforeRange,
+        let afterRange = currentAfterRange else {
+      return
+    }
+
+    let hunk = Hunk(beforeRange: beforeRange, afterRange: afterRange, contents: contents)
+    beforeHunks[beforeFile, default: []].append(hunk)
+    afterHunks[afterFile, default: []].append(hunk)
+
+    currentBeforeRange = nil
+    currentAfterRange = nil
+    contents.removeAll()
+  }
+
+  diffString.enumerateLines { line, _ in
+    // Extract file
+    // Example:
+    // --- a/components/AppBar/src/MDCAppBarContainerViewController.m
+    if let fileParts = line.split(around: "diff --git a/").1 {
```

Comment:
Is `.split(around: )` new?

Reply:
Ah. Looks like I unwittingly pulled this in from the Swift PM target.